src/dune_pkg/opam_solver.ml

```diff
@@ -351,15 +351,20 @@ let filter_to_blang ~package ~loc filter =
   filter_to_blang filter
 ;;
 
-let simplify_filter =
+let simplify_filter solver_env =
   OpamFilter.partial_eval (fun var ->
     match OpamVariable.Full.scope var with
     | Global ->
-      let name = OpamVariable.Full.variable var in
-      if OpamVariable.equal name (OpamVariable.of_string "with-doc")
-         || OpamVariable.equal name (OpamVariable.of_string "with-test")
-      then Some (B false)
-      else None
+      let name = OpamVariable.Full.variable var |> Variable_name.of_opam in
+      if Variable_name.(equal name with_test)
```

Comment:
A bit nitpicky but `name` could be shadowed if `Variable_name` introduces `name` (which doesn't sound all that unlikely). There would probably be a warning, but I feel writing out `Variable_name.equal` and `Variable_name.with_test` makes it easier to understand.